omod/src/main/java/org/openmrs/module/adminui/page/controller/myaccount/ChangeDefaultsPageController.java

```diff
@@ -54,25 +70,68 @@ public String post(PageModel model, @BindParams UserDefaults userDefaults,
                     UiCommonsConstants.SESSION_ATTRIBUTE_ERROR_MESSAGE, "adminui.account.defaults.fail");
             return "account/changeDefaults";
         }
-
-        model.addAttribute("userDefaults", userDefaults);
-        model.addAttribute("locales", Context.getAdministrationService().getPresentationLocales());
         return "myaccount/myAccount";
     }
 
-    public class UserDefaults {
+    public class LocaleOption {
 
-        private String defaultLocale;
-        private String proficientLocales;
+        private Locale locale;
+        private boolean selected;
+        private String name;
+        private String language;
 
-        public UserDefaults() {
+        public LocaleOption() {
         }
 
-        public UserDefaults(String defaultLocale, String proficientLocales) {
-            this.defaultLocale = defaultLocale;
-            this.proficientLocales = proficientLocales;
+        public LocaleOption(Locale locale, boolean selected, String name, String language) {
+            this.locale = locale;
+            this.selected = selected;
+            this.name = name;
+            this.language = language;
+        }
+
+        public Locale getLocale() {
+            return locale;
+        }
+
+        public void setLocale(Locale locale) {
+            this.locale = locale;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public void setName(String name) {
+            this.name = name;
+        }
+
+        public boolean getSelected() {
+            return selected;
+        }
+
+        public void setSelected(boolean selected) {
+            this.selected = selected;
+        }
+
+        public String getLanguage() {
+            return language;
         }
 
+        public void setLanguage(String language) {
+            this.language = language;
+        }
+    }
+    
+    public class UserDefaults {
+
+        private String defaultLocale;
+        private Set<LocaleOption> proficientLocalesSet;
```

Comment:
The options to select a default locale and proficient locales should be the same list  i.e the ones returned by AdministrationService.getPresentationLocales()